packages/devtools_app/lib/src/analytics/analytics_common.dart

```diff
@@ -23,3 +23,15 @@
 /// Then, add your fields to the `GtagEventDevTools.withScreenMetrics` factory
 /// constructor.
 abstract class ScreenAnalyticsMetrics {}
+
+/// Extracts the current DevTools page from the given [url].
+String extractCurrentPageFromUrl(String url) {
```

Comment:
let's move this to url_utils.dart, and then you can add a test to url_utils_test.dart